lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -804,6 +810,19 @@ newtype MinWithdrawal = MinWithdrawal
 instance FromHttpApiData MinWithdrawal where
     parseUrlPiece = bimap (T.pack . getTextDecodingError) MinWithdrawal . fromText
 
+instance FromHttpApiData ApiRelativeAddressIndex where
+    parseUrlPiece = bimap (T.pack . getTextDecodingError) ApiRelativeAddressIndex . fromText
+
+instance FromText Word31 where
```

Comment:
This could go into the text-class package I think, and perhaps it could be defined in terms of another typeclass like Integral.

Reply:
done, although now it is Word32. After the rebase many things need to change